src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -18,7 +18,7 @@ public abstract partial class Workspace
         private readonly Dictionary<ProjectId, ISet<DocumentId>> _projectToOpenDocumentsMap = new Dictionary<ProjectId, ISet<DocumentId>>();
 
         // text buffer maps
-        private readonly Dictionary<SourceTextContainer, DocumentId> _bufferToDocumentInCurrentContextMap = new Dictionary<SourceTextContainer, DocumentId>();
+        private readonly Dictionary<SourceTextContainer, OneOrMany<DocumentId>> _bufferToDocumentInCurrentContextMap = new Dictionary<SourceTextContainer, OneOrMany<DocumentId>>();
```

Comment:
@dpoeschl That actually sounds more cleaner, let me split this into 2 maps to make the active context tracking more explicit.